### PR TITLE
Move to use glide instead of godeps

### LINKED ISSUE #630: Use glide instead of Godeps
This will make it possible to *not* check in the massive `vendor` directory.

### PR DESCRIPTION
**Pull request by lpabon**
This is required because we needed to update the Kubernetes
APIs to use 1.6+. The biggest issue is that Kuberentes pulled
a lot of the APIs out of the tree and moved them to two different
repos: apimachinery, and client-go. This would have been a nightmare
to do in GoDeps, so I finally decided to do this in glide.

This will enable a new change for development, allowing '--kubeconfig'
to be passed to Heketi, enabling it to connect to a Kuberentes cluster
from the developers laptop.

I'm no git expert, so I couldn't figure out how to make the hated
vendor directory go into its own git commit, sorry for the pain.

Closes #630

Signed-off-by: Luis Pabón <luis.pabon@coreos.com>

### CODE REVIEW

**Comment by lpabon** [comment]
Closes #455

File: executors/kubeexec/kubeexec_test.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 	"os"
 	"testing"
 
-	"k8s.io/kubernetes/pkg/client/restclient"
+	restclient "k8s.io/client-go/rest"
```
Comment (MohamedAshiqrh) [inline executors/kubeexec/kubeexec_test.go]: nice.

File: extras/docker/fromsource/Dockerfile [hunk 2]
```diff
@@ -11,6 +11,8 @@ ENV GOPATH=$BUILD_HOME/golang
 ENV PATH=$GOPATH/bin:$PATH
 ENV HEKETI_BRANCH="master"
 
+RUN curl https://glide.sh/get | sh
+
 # install dependencies, build and cleanup
 RUN mkdir $BUILD_HOME $GOPATH && \
```
Comment (MohamedAshiqrh) [inline extras/docker/fromsource/Dockerfile]: This is not a big change but we can reduce a layer.

RUN curl https://glide.sh/get | sh && \
mkdir $BUILD_HOME $GOPATH && \ 
.
.
.

**Comment by MohamedAshiqrh** [comment]
This is good. I verified this and works as expected for me. I deleted the vendor directory and then did glide install -v. it was showing some packages missing before that.

**Comment by obnoxxx** [comment]
First comments:

1. It is indeed good to see checked-in vendor/ go.
2. This is not documented. I had to look into travis.yml to find out that I had to do "dnf install glide" followed by "glide install -v" in order to be able to run make... ==> it needs to be documented. 
3. The fact that make does not work out of the box is unfortunate. I modified the Makefile to run glide install before running go build, but we should do it more cleverly, because this way, the time for make is longer. Can we detect whether glide install has been run? Is there a way to update the "glide install"?
4. I also experimented to have make use "go build -i" instead of "go build" to reduce the time needed for "make". Running "glide install -v" prior to "go make -i" greatly reduced the usefulness of "-i". :-(

**Comment by lpabon** [comment]
@obnoxxx Great points. I'll make those changes.

**Comment by raghavendra-talur** [comment]
@obnoxxx I don't understand your comment "It is indeed good to see checked-in vendor/ go."

@lpabon Changes look good.

**Comment by obnoxxx** [comment]
@raghavendra-talur i wanted to express that I don't like the checked-in vendor directory. And I like the fact that this patch removes it.

**Comment by lpabon** [comment]
@obnoxxx @raghavendra-talur @MohamedAshiqrh It works now just by typing `make`. I changed the Travis instructions to use this method from now on.

**Comment by obnoxxx** [comment]
@lpabon looks good.

The only thing I don't like in the makefile is the download of glide.
I don't like flooding of the system with manual dowloads for something
that can also be installed via system packages (rpms / debs / ...).

Apart from that: Nice!

**Comment by obnoxxx** [comment]
I'm inclined to merge this PR as is, and think about a possible follow-up patch for the makefile to address my above comment. I like to have this in rather now than late. :-)

**Comment by obnoxxx** [comment]
LGTM

**Pull request merged**